interfaces/devdoc/file_requirements.md

```diff
@@ -0,0 +1,116 @@
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_WRITE_ASYNC_RESULT, file_write_async, FILE_HANDLE, handle, CONSTBUFFER_HANDLE, source, uint64_t, position, FILE_WRITE_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+```
+
+`file_write_async` issues an asynchronous write request.
+
+**SRS_FILE_43_008: [** `file_write_async` writes `source`'s content to the position offset in the file and calls `user_call_back` passing `user_context` and success.**]**
+
+**SRS_FILE_43_009: [** If `handle` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_010: [** If `source` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_012: [** If `user_callback` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_013: [** If writing `size` bytes at `position` would exceed `desired_file_size` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_014: [** `file_write_async` shall create an event to be used for the asynchronous write operation. **]**
+
+**SRS_FILE_43_015: [** If there are any failures, `file_write_async` shall fail and return `FILE_WRITE_ASYNC_ERROR`. **]**
+
+**SRS_FILE_43_030: [** `file_write_async` shall succeed and return `FILE_WRITE_ASYNC_OK`. **]**
+
+## file_read_async
+
+```c
+MOCKABLE_FUNCTION_WITH_RETURNS(, int, file_read_async, FILE_HANDLE, handle, uint32_t, size, uint64_t, position, FILE_READ_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+```
+
+`file_read_async` issues an asynchronous read request.
+
+**SRS_FILE_43_016: [** `file_read_async` reads a byte array identified by `destination`, `size` from `position` and calls `user_callback` passing `user_context`. **]**
+
+**SRS_FILE_43_017: [** If `handle` is `NULL` then `file_read_async` shall fail and return `FILE_READ_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_019: [** If `position` + `size` would exceed the available bytes in the file then `file_read_async` shall fail and return `FILE_READ_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_020: [** If `user_callback` is `NULL` then `file_read_async` shall fail and return `FILE_READ_ASYNC_INVALID_ARGS`. **]**
+
+**SRS_FILE_43_021: [** `file_read_async` shall create an event to be used for the asynchronous read operation. **]**
+
+**SRS_FILE_43_032: [** `file_read_async` shall allocate `size` bytes for reading the data. **]**
+
+**SRS_FILE_43_022: [** If there are any failures then `file_read_async` shall fail and return `FILE_READ_ASYNC_ERROR`. **]**
+
+**SRS_FILE_43_031: [** `file_read_async` shall succeed and return `FILE_READ_ASYNC_OK`. **]**
+
+## file_extend_filesize
+
+```c
+MOCKABLE_FUNCTION_WITH_RETURNS(, int, file_extend_filesize, FILE_HANDLE, handle, uint64_t, desired_size, bool, has_manage_volume)(0, MU_FAILURE);
+```
+
+**SRS_FILE_43_023: [** `file_extend_filesize` resizes the file to `desired_size`. **]**

```

Comment:
>**SRS_FILE_43_023: [**  [](start = 0, length = 23)

Can probably remove the SRS part from here, this is more of the description of the function.